src/common/actions/snaps.js

```diff
@@ -42,6 +42,20 @@ export function fetchUserSnaps(owner) {
   };
 }
 
+function shouldFetchUserSnaps(state) {
+  const snaps = state.snaps;
+
+  return !snaps.isFetching;
+}
+
+export function fetchUserSnapsIfNeeded(owner) {
+  return function(dispatch, getState) {
+    if (shouldFetchUserSnaps(getState())) {
```

Comment:
Can you show me the pattern we're using here? The pattern I've used is from redux docs, and also what I used in the prototype app before mu https://github.com/reactjs/redux/blob/master/examples/async/src/actions/index.js#L46